lib/game.rb

```diff
@@ -0,0 +1,78 @@
+class Game
+
+  include Players
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+                      [0,1,2],
+                      [3,4,5],
+                      [6,7,8],
+                      [0,3,6],
+                      [1,4,7],
+                      [2,5,8],
+                      [0,4,8],
+                      [2,4,6]
+                              ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    if board.turn_count % 2 == 0
+      player_1
+    else
+      player_2
+    end
+  end
+
+  def turn
+    move = current_player.move(board)
+    if board.valid_move?(move)
+      board.update(move, current_player)
+    else
+      puts "Your input was invalid."
+      turn
+    end
+  end
+
+  def play
+    while !(over?)
+      turn
+      board.display
+    end
+    if won?
+      puts "Congratulations #{winner}!"
+    else draw?
+      puts "Cats Game!"
+    end
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |index|
+    position_1 = board.cells[index[0]]
+    position_2 = board.cells[index[1]]
+    position_3 = board.cells[index[2]]
+      if position_1 == "X" && position_2 == "X" && position_3 == "X" || position_1 == 'O' && position_2 == 'O' && position_3 == 'O'
+        return index
+      end
+    end
+    false
+  end
+
+  def draw?
+    board.full? && !(won?)
+  end
+
+  def over?
+    won? || draw?
+  end
+
+  def winner
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.